lib/Doctrine/ORM/Proxy/Factory/StaticProxyFactory.php

```diff
@@ -175,6 +196,44 @@ private function transientFieldsFqns(ClassMetadata $metadata) : array
         return $transientFieldsFqns;
     }
 
+    private function collectionFieldsFqns(ClassMetadata $metadata) : array
+    {
+        $transientFieldsFqns = [];
+
+        foreach ($metadata->getDeclaredPropertiesIterator() as $name => $property) {
+            if (! $property instanceof ToManyAssociationMetadata) {
```

Comment:
I don't like this kind of filtering of an iterator - do want a dedicated metadata API endpoint

Reply:
Thinking about how to avoid ton of getDeclared*() methods, quick idea: what would you say to PropertyFilterIterator + binary flags to pick your own set + factory methods (forAssociations())?

Reply:
Bad idea: lots of untyped methods then. Having multiple dedicated accessors for metadata isn't a problem IMO.